src/_includes/layouts/post.njk

```diff
@@ -1,8 +1,14 @@
 ---
 layout: layouts/base.njk
 type: post
+eleventyComputed:
+    permalink: "{% if date | isFuturePost %}false{% else %}{{permalink}}{% endif %}"
```

Comment:
By using a collection I don't think you need this anymore?

Reply:
It's necessary, if I don't set the permalink to false it will certainly be excluded from the collection, but it will still generate the post page in production.